docs/src/content/docs/challenges/forms/40-control-value-accessor.md

```diff
@@ -0,0 +1,20 @@
+---
+title: 🟠 Control Value Accessor
+description: Challenge 40 is about creating a custom form control that implements Control Value Accessor interface.
+author: stanislav-gavrilov
+challengeNumber: 40
+command: forms-control-value-accessor
+sidebar:
+  order: 119
+  badge: New
+---
+
+:::note
+WIP: The following documentation will be reviewed and improved. However, you can still take on the challenge. If you don’t understand a certain part, please feel free to reach out or create an issue.
+:::
+
+## Information
+
+In this exercise, you have to rewrite FeedbackControlComponent. It should implement Control Value Accessor interface.
+This interface is crucial for creating custom form controls that can interact seamlessly with Angular's forms API.
+The primary goal is to use control in the feedbackForm to eliminate the need for using @Output to retrieve the value and check it.
```

Comment:
I propose the following description which can be further improved. 

## Information

In this challenge, the goal is to create a custom form field that is using the Form API of Angular `ControlValueAccessor`. You can find the documentation [here](https://angular.io/api/forms/ControlValueAccessor). This interface is crucial for creating custom form controls that can interact seamlessly with Angular's forms API.

## Statement

The primary goal is to use control in the feedbackForm to eliminate the need for using @Output to retrieve the value and inject it into the `FormGroup`.

Currently, rating is coded this way:

```html
  <app-feedback-control (ratingUpdated)="rating = $event">
  </app-feedback-control>
```

```ts
  rating: string | null = null;

  submitForm(): void {
    this.feedBackSubmit.emit({
      ...this.feedbackForm.value,
      rating: this.rating, // not inside the `FormGroup` and no validation
    });
  }
```

The goal is to include rating into the `FormGroup` 

```html
  <app-feedback-control [formControl]="feedbackForm.controls.rating">
  </app-feedback-control>
```

Reply:
I took your proposal and added info about validation as @jdegand asked. Please check if everything is ok.

Reply:
also, i renamed app-feedback-control to app-rating-control, as it feels more suitable to me.

Reply:
Yes validation is a good point. I thought of it but I didn't mention it because it will be inside another challenge as well.